<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Using the dccvalidator app in VEO-IBD</title>






<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Using the dccvalidator app in VEO-IBD</h1>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>We’ve built the dccvalidator tool to streamline the process of data validation and QA/QC. This application performs many of the routine data quality checks that were previously conducted by hand, with the hopes that it will help you, the data contributor, get your data checked, validated, and shared easily and quickly.</p>
<p>The application is hosted on a Shiny server <a href="https://shinypro.synapse.org/users/cmolitor/dccvalidator">here</a>.</p>
<div id="application-requirements" class="section level2">
<h2>Application Requirements</h2>
<p>To use this application you must:</p>
<ol style="list-style-type: decimal">
<li>Be logged in to Synapse in your browser</li>
<li>Be a <a href="https://docs.synapse.org/articles/accounts_certified_users_and_profile_validation.html">Synapse certified user</a></li>
<li>Be a member of the <a href="https://www.synapse.org/#!Team:3382930">VEO-IBD team</a></li>
</ol>
<p>Some portions of the app submit data to Synapse. This allows curators at Sage to troubleshoot issues if needed. No one outside the Sage VEO-IBD curation team will be able to download the data.</p>
</div>
<div id="data-sharing-agreement" class="section level2">
<h2>Data Sharing Agreement</h2>
<p>In order to contribute data to the VEO-IBD Synapse project, you are required to sign the VEO-IBD Data Sharing Agreement.</p>
<ul>
<li><p>If you discover you have released sensitive information in error, please follow the <a href="https://docs.synapse.org/articles/contribute_and_access_controlled_use_data.html#flagging-inappropriate-data-use">process in Synapse to flag</a> affected data.</p></li>
<li><p>We are also interested in rapidly identifiying data that does not conform to our quality control standards. If you discover data is contaminated or sample identity is questionable due to a concordance analysis, for example, please reach out to the curation team -<a href="mailto:VEOIBD_SageAdmin@synapse.org">VEOIBD_SageAdmin@synaspse.org</a>.</p></li>
</ul>
</div>
<div id="terminology" class="section level2">
<h2>Terminology</h2>
<div id="what-is-a-biospecimen" class="section level4">
<h4>What is a biospecimen?</h4>
<p>A biospecimen is a sample of material such as tissue, cells, DNA, RNA or protein that has a unique identifier associated with it. The same biospecimen may be characterized in multiple assay types. In this case, the unique identifier should remain the same. We strongly recommend you do <em>not</em> name specimens using individual identifiers. In the case where multiple sequencing libaries are prepared from a single biospecimen, <code>LibraryID</code> is an available key. Replicates are tracked using integers and the keys <code>technicalReplicate</code> and <code>sequencingReplicate</code>.</p>
</div>
<div id="what-is-a-manifest-how-is-a-manifest-different-than-a-metadata-file" class="section level4">
<h4>What is a manifest? How is a manifest different than a metadata file?</h4>
<p>A manifest is .tsv or .txt file with data files to be uploaded to Synapse as entries in each row. Details of a manifest are described in the <a href="https://docs.synapse.org/articles/uploading_in_bulk.html">Uploading and Downloading Data in Bulk</a> Synapse User Guide. While a metadata file will be stored on Synapse as a flat file, and select variables added as file annotations, all variables in a manifest file will live as annotations respective to the file in that row. To successfully upload a file, you must specify the local <code>path</code> to the file and the Synapse ID of the folder in the <code>parent</code> column.</p>
</div>
<div id="can-i-track-relationships-between-files-in-synapse-as-i-upload-data" class="section level4">
<h4>Can I track relationships between files in Synapse as I upload data?</h4>
<p>Yes, Synapse supports <a href="https://docs.synapse.org/articles/provenance.html">Provenance</a>! Provenance can be leveraged to connect raw data to reprocessed or summarized data. Populate the <code>used</code> column in the manifest with the synID. The required values format for linking multiple files is <code>used = synID;synID</code>.</p>
</div>
<div id="can-i-associate-relevant-code-to-files" class="section level4">
<h4>Can I associate relevant code to files?</h4>
<p>Yes, with <a href="https://docs.synapse.org/articles/provenance.html">Provenance</a>. Populate the <code>executed</code> column with the url to your Github repo.</p>
</div>
</div>
<div id="study-and-assay-documentation-upload" class="section level2">
<h2>Study and assay documentation upload</h2>
<p>Each study in PEC will have accompanying documentation in the PEC portal. Here is an example of study <a href="https://adknowledgeportal.synapse.org/#/Explore/Studies?Study=syn8391648">documentation in the Accelerating Medicines Partnership in Alzheimer’s Disease portal</a>, developed by Sage Bionetworks.</p>
<p>You can submit your documentation through the dccvalidator app on the Documentation page. There should be a study description for the whole study, and an assay description for each of the assays that was performed. These can be in a single file, or you can upload multiple files to the assay description section.</p>
<div id="how-do-i-get-access-to-a-staging-folder-to-upload-data" class="section level4">
<h4>How do I get access to a Staging folder to upload data?</h4>
<p>With a new study, there may not yet be a <strong>Staging folder</strong> in the PEC Knowledge Portal. Please contact us - <a href="mailto:VEOIBD_SageAdmin@synapse.org">VEOIBD_SageAdmin@synaspse.org</a>.</p>
</div>
</div>
<div id="data-validation" class="section level2">
<h2>Data validation</h2>
<div id="metadata-requirements" class="section level3">
<h3>Metadata requirements</h3>
<p>Each study should include metadata that would help a new researcher understand and reuse the data. In most cases, we will expect 4 files:</p>
<ol style="list-style-type: decimal">
<li><strong>Individual metadata</strong> describing each individual in the study
<ul>
<li>Each row corresponds to a unique individual</li>
</ul></li>
<li><strong>Biospecimen metadata</strong> describing the specimens that were collected</li>
<li><strong>Assay metadata</strong> describing the assay that was performed. If multiple assays were part of the study, there will be one assay file for each.</li>
<li>A <strong>manifest</strong> listing each file that will be uploaded. Remember to include your metadata files in the manifest.</li>
</ol>
<p>Metadata file templates are <a href="https://www.synapse.org/#!Synapse:syn20729790">available in the VEO-IBD Synapse project</a>.</p>
<p>If you don’t see a template for the assay(s) in your study, please send a request for a new schema to <a href="mailto:VEOIBD_SageAdmin@synapse.org">VEOIBD_SageAdmin@synapse.org</a>. We depend on your expertise to develop schemas that capture the most pertinent metadata!</p>
</div>
<div id="validating-metadata" class="section level3">
<h3>Validating metadata</h3>
<p>The data validation portion of the app allows you to upload metadata files (as .csv) and the manifest (as .tsv or .txt) and view the results of a series of automated checks.</p>
<p>Examples of the types of checks we perform are: - All required columns from the templates are present - Individuals and specimens have unique identifiers - Metadata terms conform to a controlled vocabulary, where applicable</p>
</div>
<div id="viewing-data-summary" class="section level3">
<h3>Viewing data summary</h3>
<p>We also provide a summary of the files you have uploaded, showing the number of individuals, specimens, and files. We visualize the data in each column by its data type to help spot unexpected missing values.</p>
</div>
</div>
<div id="types-of-data-to-upload" class="section level2">
<h2>Types of data to upload</h2>
<p>The <a href="https://www.synapse.org/#!Synapse:syn20959298">Data Analysis Core</a> will reprocess common data types during PEC Phase II. Currently, common data types are RNASeq, ATACSeq, ChipSeq and all single cell data. For common data types, <strong>only fastq files are required</strong>. For other data types, please provide <strong>fastq</strong> and <strong>bam</strong> files.</p>
<div id="raw" class="section level4">
<h4>Raw</h4>
<ul>
<li><strong>fastq</strong> - All lanes and barcodes of the same replicate are merged into one file per paired-end. Therefore, each specimenID will be associate to at least two files for each paired-end read. We recommend to include specimenID in the filename.</li>
</ul>
</div>
<div id="processed" class="section level4">
<h4>Processed</h4>
<ul>
<li><p><strong>count matricies</strong> - RNASeq, ChIPSeq and ATACSeq data all produce count matricies. These file types are especially useful for data users who want to compare their own datasets without bioinformatic processing.</p></li>
<li><p><strong>peaks</strong> - A processed data type specific to ChIPSeq output.</p></li>
</ul>
</div>
</div>
<div id="uploading-data-to-synapse-after-validation" class="section level2">
<h2>Uploading data to Synapse after validation</h2>
<p>Once data has passed validation, and the PEC data curators permit edit permissions to the <strong>Staging</strong> folder, you will use your newly created manifest file to upload data using <code>syncToSynapse</code>. You can execute <code>syncToSynapse</code> in the <a href="https://python-docs.synapse.org/build/html/synapseutils.html#synapseutils.sync.syncToSynapse">Python client</a> and <a href="https://github.com/Sage-Bionetworks/synapserutils#upload-data-in-bulk">R client</a>. The Synapse Python client supports multithreaded upload and will provide faster upload speeds than the Synapse R client. For getting started with the Synapse programmatic clients, please visit our <a href="https://docs.synapse.org/articles/api_documentation.html">Synapse docs</a>.</p>
</div>
<div id="data-release" class="section level2">
<h2>Data Release</h2>
<p>Data is uploaded to a <strong>Staging</strong> folder, private to each individual group. Once curated, data is moved to a <strong>VEO-IBD</strong> folder for a limited period of time where all consortium members have access to the data via the VEO-IBD Team. Finally, data is made public to Synapse users in a <strong>Data</strong> folder. All data upload takes place in the <a href="https://www.synapse.org/#!Synapse:syn25541843/wiki/609567">VEO-IBD Synapse project</a>. While access to the project is public, restrictions are associated with the <strong>Staging</strong> and <strong>VEO-IBD</strong> folder to make sure the data remains private for the appropriate period of time.</p>
<p>Synapse IDs are always preserved (i.e. IDs remain associated to the file).</p>
</div>
<div id="get-support" class="section level2">
<h2>Get support</h2>
<p>Please send questions to <a href="mailto:VEOIBD_SageAdmin@synapse.org">VEOIBD_SageAdmin@synapse.org</a>.</p>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
